exoscale/api/compute.py

```diff
@@ -0,0 +1,2088 @@
+
+    ### Elastic IP
+
+    def create_elastic_ip(
+        self,
+        zone,
+        healthcheck_mode=None,
+        healthcheck_port=None,
+        healthcheck_path="/",
+        healthcheck_interval=10,
+        healthcheck_timeout=2,
+        healthcheck_strikes_ok=3,
+        healthcheck_strikes_fail=2,
+    ):
+        """
+        Create an Elastic IP.
+
+        Parameters:
+            zone (Zone): the zone in which to create the Elastic IP
+            healthcheck_mode (str): optional healthcheck mode
+            healthcheck_port (int): healthcheck port,
+                required if healthchecking is enabled
+            healthcheck_path (str): healthcheck probe HTTP request path,
+                required in "http" mode
+            healthcheck_interval (int): probe interval in seconds
+            healthcheck_timeout (int): time in seconds before
+                considering a probe failed, must be lower than interval
+            healthcheck_strikes_ok (int): number of successful probes
+                before considering the target healthy
+            healthcheck_strikes_fail (int): number of unsuccessful probes
+                before considering the target unhealthy
+
+        Returns:
+            ElasticIP: the Elastic IP created
+        """
+
+        # Unset healthcheck default values if healthchecking is not enabled
+        if healthcheck_mode is None:
+            healthcheck_path = None
+            healthcheck_interval = None
+            healthcheck_timeout = None
+            healthcheck_strikes_ok = None
+            healthcheck_strikes_fail = None
+
+        try:
+            # We have to pass function arguments using **kwargs form because
+            # of the hyphen in "strikes-ok"/"strikes-fail" arguments.
+            res = self.cs.associateIpAddress(
+                **{
+                    "zoneid": zone.id,
+                    "mode": healthcheck_mode,
+                    "port": healthcheck_port,
+                    "path": healthcheck_path,
+                    "interval": healthcheck_interval,
+                    "timeout": healthcheck_timeout,
+                    "strikes-ok": healthcheck_strikes_ok,
+                    "strikes-fail": healthcheck_strikes_fail,
+                }
+            )
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+        return ElasticIP.from_cs(self, res["ipaddress"])
+
+    def list_elastic_ips(self, zone=None, **kwargs):
+        """
+        List Elastic IPs.
+
+        Parameters:
+            zone (Zone): a zone to restrict results to
+            
+        Yields:
+            ElasticIP: the next Elastic IP
+        """
+
+        try:
+            _list = self.cs.listPublicIpAddresses(
+                fetch_list=True, zoneid=getattr(zone, "id", None), **kwargs
+            )
+
+            for i in _list:
+                if not i["iselastic"]:
+                    continue
```

Comment:
You can ask the API to filter directly with `iselastic=True`

Reply:
Cool, thanks! Fixed in f3777d8